pkg/cluster/px/px.go

```diff
@@ -1402,6 +1406,11 @@ func (ops *pxClusterOps) checkAPIDaemonset(namespace string) error {
 			},
 		}
 
+		if affinity != nil && affinity.NodeAffinity != nil {
+			apiDS.Spec.Template.Spec.Affinity = &v1.Affinity{
```

Comment:
Why not set the complete affinity instead of just node affinity?

Reply:
done